finagle-thrift/src/main/scala/com/twitter/finagle/thrift/ThriftClientFramedCodec.scala

```diff
@@ -28,41 +28,60 @@ object ThriftClientFramedCodec {
 class ThriftClientFramedCodecFactory(
     clientId: Option[ClientId],
     _useCallerSeqIds: Boolean,
-    _protocolFactory: TProtocolFactory)
+    _protocolFactory: TProtocolFactory,
+    _attemptProtocolUpgrade: Boolean)
   extends CodecFactory[ThriftClientRequest, Array[Byte]]#Client {
 
-  def this(clientId: Option[ClientId]) = this(clientId, false, Protocols.binaryFactory())
+  def this(clientId: Option[ClientId], _useCallerSeqIds: Boolean, _protocolFactory: TProtocolFactory) =
+    this(clientId, _useCallerSeqIds, _protocolFactory, true)
+
+  def this(clientId: Option[ClientId]) =
+    this(clientId, false, Protocols.binaryFactory(), true)
 
   def this(clientId: ClientId) = this(Some(clientId))
 
   // Fix this after the API/ABI freeze (use case class builder)
   def useCallerSeqIds(x: Boolean): ThriftClientFramedCodecFactory =
-    new ThriftClientFramedCodecFactory(clientId, x, _protocolFactory)
+    new ThriftClientFramedCodecFactory(clientId, x, _protocolFactory, _attemptProtocolUpgrade)
 
   /**
-   * Use the given protocolFactory in stead of the default `TBinaryProtocol.Factory`
+   * Use the given protocolFactory instead of the default `TBinaryProtocol.Factory`
    */
   def protocolFactory(pf: TProtocolFactory) =
-    new ThriftClientFramedCodecFactory(clientId, _useCallerSeqIds, pf)
+    new ThriftClientFramedCodecFactory(clientId, _useCallerSeqIds, pf, _attemptProtocolUpgrade)
+
+  /**
+   * Enable or disable protocol upgrading.
+   */
+  def attemptProtocolUpgrade(enabled: Boolean): ThriftClientFramedCodecFactory =
```

Comment:
Let's name that `withNoProtocolUpgrade` as per [item 2 in design principles](http://twitter.github.io/finagle/guide/Configuration.html#design-principles). But please, leave the stack param as is.